pay-api/migrations/versions/2025_07_30_38045c22fe00_gst_data_model_changes.py

```diff
@@ -0,0 +1,81 @@
+    sa.Column('rate', sa.Numeric(precision=6, scale=4), nullable=False, comment='Tax rate as decimal, e.g. 0.0500 for 5%'),
+    sa.Column('start_date', sa.DateTime(timezone=True), nullable=False, comment='When this tax rate becomes effective'),
+    sa.Column('effective_end_date', sa.DateTime(timezone=True), nullable=True, comment='When this tax rate expires'),
+    sa.Column('description', sa.String(length=200), nullable=True, comment='Description of the tax rate'),
+    sa.Column('created_on', sa.DateTime(), nullable=False),
+    sa.Column('updated_on', sa.DateTime(), nullable=True),
+    sa.Column('created_by', sa.String(length=50), nullable=False),
+    sa.Column('created_name', sa.String(length=100), nullable=True),
+    sa.Column('updated_by', sa.String(length=50), nullable=True),
+    sa.Column('updated_name', sa.String(length=50), nullable=True),
+    sa.PrimaryKeyConstraint('id')
+    )
+    with op.batch_alter_table('distribution_codes', schema=None) as batch_op:
+        batch_op.add_column(sa.Column('statutory_fees_gst_distribution_code_id', sa.Integer(), nullable=True, comment='Distribution code for GST on statutory fees'))
+        batch_op.add_column(sa.Column('service_fee_gst_distribution_code_id', sa.Integer(), nullable=True, comment='Distribution code for GST on service fees'))
+        batch_op.create_foreign_key(None, 'distribution_codes', ['service_fee_gst_distribution_code_id'], ['distribution_code_id'])
+        batch_op.create_foreign_key(None, 'distribution_codes', ['statutory_fees_gst_distribution_code_id'], ['distribution_code_id'])
+
+    with op.batch_alter_table('distribution_codes_history', schema=None) as batch_op:
+        batch_op.add_column(sa.Column('statutory_fees_gst_distribution_code_id', sa.Integer(), autoincrement=False, nullable=True, comment='Distribution code for GST on statutory fees'))
+        batch_op.add_column(sa.Column('service_fee_gst_distribution_code_id', sa.Integer(), autoincrement=False, nullable=True, comment='Distribution code for GST on service fees'))
+
+    with op.batch_alter_table('fee_schedules', schema=None) as batch_op:
+        batch_op.add_column(sa.Column('enable_gst', sa.Boolean(), nullable=True, comment='Flag to indicate if GST should be calculated for this fee schedule'))
+
+    with op.batch_alter_table('invoices', schema=None) as batch_op:
+        batch_op.add_column(sa.Column('gst', sa.Numeric(precision=19, scale=2), nullable=True, comment='Total GST amount including statutory and service fees GST'))
+
+    with op.batch_alter_table('payment_line_items', schema=None) as batch_op:
+        batch_op.add_column(sa.Column('statutory_fees_gst', sa.Numeric(precision=19, scale=2), nullable=True, comment='GST for statutory fees (filing_fees)'))
+        batch_op.add_column(sa.Column('service_fees_gst', sa.Numeric(precision=19, scale=2), nullable=True, comment='GST for service fees'))
+
```

Comment:
do we need:
ALTER TABLE corp_types ADD COLUMN enable_gst_split BOOLEAN DEFAULT FALSE;